sample_factory/algo/sampling/batched_sampling.py

```diff
@@ -266,6 +267,9 @@ def advance_rollouts(self, policy_id: PolicyID, timing) -> Tuple[List[Dict], Lis
 
             # record the results from the env step
             rewards_cpu = rewards.cpu()
+            
+            if isinstance(infos, list):
+                infos = list_of_dicts_to_dict_of_lists(infos)
```

Comment:
Actually... I see a problem with this.
The way this list_of_dicts... function is written, it will append to a list if and only if the key is present in the individual dictionary.

So if environments 1 and 3 have a key in info and environment 2 does not, we will end up with lists of length 2.
And so information for env 3 will correspond to environment 2, which will cause bugs.

Unfortunately we just can't rely on the fact that envs will return a consistent set of keys every time.
VizDoom does not. Lots of environments return a special set of keys on the last step of the episode.

Bottom line, we can't use it here.
I actually fixed the crash inside of process_rewards by this:
`time_outs = None if not isinstance(infos, dict) else infos.get('time_outs')` 
If this fixes your problem, maybe let's just use this for now?